toolkit/components/uniffi-bindgen-gecko-js/src/templates/js/Enum.jsm

```diff
@@ -0,0 +1,63 @@
+{%- let enum_ = ci.get_enum_definition(name).unwrap() %}
+
+class {{ enum_.nm() }} {}
+
+EXPORTED_SYMBOLS.push("{{ enum_.nm() }}");
+
```

Comment:
I think we should create a base class for the enum itself.  That way:
  - Users can use `instanceof` to check if an object is any of the variants.
  - We can make the variants nested classes, which I think is normally what users want (for example `DeviceType.Mobile`)